util/util.go

```diff
@@ -83,6 +83,22 @@ func MergeParameters(params ...[]wfv1.Parameter) []wfv1.Parameter {
 	return resultParams
 }
 
+// MergeArtifacts merges artifact argument slices
+// Merge the slices based on arguments order (first is high priority).
+func MergeArtifacts(artifactSlices ...[]wfv1.Artifact) []wfv1.Artifact {
+	var result []wfv1.Artifact
+	alreadyMerged := make(map[string]bool)
+	for _, artifacts := range artifactSlices {
+		for _, item := range artifacts {
+			if _, found := alreadyMerged[item.Name]; !found {
```

Comment:
minor: with a bool map, missing entries return false, so you can do this

```golang
if !alreadyMerged[item.Name] {
```

Reply:
Fixed. Thanks